### ISSUE TITLE
Resource search causes UI to reload

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#10869]
In any cluster, local or a downstream, explore the cluster.

Click on the search icon in the top right and then star tying 'service', you'll see something like:

![image](https://github.com/rancher/dashboard/assets/1955897/39bb4654-8938-4417-b88b-daf7ae5fcbf0)

Select the option 'Service Accounts'.

The UI will do a hard reload, rather than navigating to that view.

This is a regression - the UI should not hard-reload.

**Comment by richard-cox** [comment]
Took a bit of tracking down, but was caused by https://github.com/rancher/dashboard/pull/10793. Reverting that commit results in no page reload.

**Comment by codyrancher** [comment]
I removed `dev-automation` from this because it looks like it's actually going to be more difficult to automate than I would've thought.

We either need to rely on globals and private implementation details of VueRouter or we need to add extra runtime code to production in order to support gathering information (route history) to determine if we reloaded or navigated to a page. 

I don't think either option is reasonable for the scope of this issue.
